src/App.js

```diff
@@ -1,45 +1,51 @@
-import logo from "./logo.svg";
 import "./App.css";
 
 //Importing Components
-import Header from "./components/Header";
-import BannerItems from "./components/Banner/BannerItems";
-import MainContent from "./components/MainContent";
-import Footer from "./components/Footer";
-import RestroPage from "./components/RestroPage";
+import Header from "./components/header";
+import MainContent from "./components/mainContent";
+import Footer from "./components/footer";
+import RestroPage from "./components/restroPage";
 import { Component } from "react";
-import CartCounterButton from "./components/CartCounterButton/CartCounterButton";
-import Checkout from "./components/Checkout/Checkout";
-
-//
+import BannerItems from "./components/bannerItems";
 
 class App extends Component {
-  constructor() {
-    super();
-    this.state = {
-      restroClickedFlag: false,
-    };
-  }
+  state = {
+    restroClickedFlag: false,
+  };
 
   restroClicked = (restro) => {
     if (restro) {
       this.setState({ restroClickedFlag: true });
     }
   };
 
+  renderHeader() {
+    return <Header />;
+  }
+
+  renderRestroDetailsAndHighlights() {
+    return (
+      <>
+        <BannerItems />
+        <MainContent restroClicked={this.restroClicked} />
+        <Footer />
+      </>
+    );
+  }
+
+  renderRestroDishesPage() {
+    {
+      return <RestroPage />;
+    }
+  }
+
   render() {
     return (
       <div className="App">
-        <Header />
-        {!this.state.restroClickedFlag && (
-          <>
-            <BannerItems />
-            <MainContent restroClicked={this.restroClicked} />
-            <Footer />
-          </>
-        )}
-        {this.state.restroClickedFlag && <RestroPage />}
-        {/*Comments*/}
+        {this.renderHeader()}
+        {!this.state.restroClickedFlag &&
+          this.renderRestroDetailsAndHighlights()}
+        {this.state.restroClickedFlag && this.renderRestroDishesPage()}
```

Comment:
You are using `restroClickedFlag` twice. It's better to destructure it at the top and then use it. 
Reason being that first - it is more readable and object lookup has a cost to it, so it's better to do the lookup once than to have to do it twice.